### ISSUE TITLE
Parsing template fails due to unsupported type: map[interface {}]interface {}

### ISSUE THREAD

**Issue by bottemav** [awslabs/aws-sam-cli#342]
Sam local is unable to parse following template.

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::Serverless-2016-10-31'

Resources:
  StateMachine:
    Type: "AWS::StepFunctions::StateMachine"
    Properties:
      StateMachineName: "SamStateMachine"
      RoleArn: !Sub "arn:aws:iam::${AWS::AccountId}:role/StateMachineServiceRole"
      DefinitionString:
        !Sub
          - |-
             {
               "Comment": "${application}",
               "StartAt": "HelloWorld",
               "States": {
                 "HelloWorld": {
                   "Type": "Pass",
                   "Result": {
                       "x-datum": 0.381018,
                       "y-datum": 622.2269926397355
                     },
                   "ResultPath": "$.coords",
                   "End": true
                 }
               }
             }
          - {
            application: "applicationName"
          }
```
When I execute `sam validate` the following error is thrown
 `invalid YAML template: json: unsupported type: map[interface {}]interface {}`.
I'm running sam local 0.2.7 on Windows 10.

**Comment by bgrindy** [comment]
I'm dealing with this issue too, same as #135

**Comment by sbuzonas** [comment]
It looks like you have your `!Sub` in the wrong place. It should be in front of a string, not a sequence.

**Comment by bottemav** [comment]
@sbuzonas You can use a sequence. The first is your string containing parameters and the second specifying the values of these parameters. See [Intrinsic Function Reference » Fn::Sub](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html#YAML)

@bgrindy Replacing `!Sub` with `Fn::Sub:` is indeed a workaround.

I don't want to close this issue as SAM Local should support `!Sub` too.

**Comment by jfuss** [comment]
The `validate` command was integrated with [SAM](https://github.com/awslabs/serverless-application-model) so this should not be an issue any longer.

Closing as this is fixed in [0.3.0](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.3.0)
